internal/statsd.go

```diff
@@ -216,5 +216,9 @@ func (sc *StatsCarrier) push(s Stat) {
 
 // Add pushes the Stat, s, onto the StatsCarrier's contribStats channel
 func (sc *StatsCarrier) Add(s Stat) {
-	sc.contribStats <- s
+	select {
+	case sc.contribStats <- s:
+	default:
```

Comment:
I'm wondering if we want to also check `<-sc.stop` here? There's a difference between `sc.contribStats` being full because `StatsCarrier.run` isn't keeping up vs being full because the `StatsCarrier` isn't running at all.